src/transformers/modeling_utils.py

```diff
@@ -3975,6 +3976,12 @@ def from_pretrained(
                     _from_pipeline=from_pipeline,
                     **kwargs,
                 )
+                if generation_config is not None:
+                    logger.info(
+                        "Both `pretrained_model_name_or_path` and `generation_config` are provided. The `generation_config`"
+                        " will be used to override the `pretrained_model_name_or_path` generation config."
+                    )
+                    model.generation_config = model.generation_config.from_dict(generation_config.to_dict())
```

Comment:
If we're going to override if `generation_config` is passed in, we can just shortcut and set this directly instead of going through `GenerationConfig.from_pretrained` first which might e.g. download files.

Reply:
I guess most configs downloaded from_pretrained are defaults and unchanged, and only a few fields are override. We probably don't want to pass in new defaults for all fields.